app/gui/view/graph-editor/src/component/visualization/container/action_bar.rs

```diff
@@ -403,14 +409,26 @@ impl ActionBar {
 
             show_reset_icon <- bool(&reset_position_icon_down,start_dragging);
             eval show_reset_icon((visibility) model.icons.set_reset_icon_visibility(*visibility));
+
+
+           // === Visualisation Chooser ===
+
+            // Note: we only want to update the chooser if it is visible, or when it becomes
+            // visible. Thus we avoid many simultaneous updates during initialisation.
+
```

Comment:
I don't understand this comment. What do you mean by "many simultaneous updates during initialization"? Why updating the chosen while it is invisible would cause many simultaneous updates?

Reply:
We update all choosers upon first receiving types from the engine (the initialization), which is mostly pointless given they are not visible (and they content may easily be changed again before it will be actually shown).

Reply:
Ok, @MichaelMauderer, would you be able to add this (^^^) information to the comment there as well, please? It would make (at least for me) it way more understandable.

Reply:
Done.